### PR TITLE
OpenLDAP - Adding support for Active Directory Auth with Start TLS

### PR DESCRIPTION
**Pull request by ShacharKidor**
**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [CIAC-2478](https://jira-hq.paloaltonetworks.local/browse/CIAC-2478).

## Description
**Support Active Directory Auth with STARTTLS as part of OpenLDAP auth integration:**

1. Added support for AD Auth (you can now login to XSOAR using active directory users). 

2. Added support for Start TLS connection. 

3. Enlarged the 'Allowed ciphers list' for SSL/TLS - In python 3.10 they hardened the SSL protocol by reducing the list of ciphers, which caused an LDAP server that did not have the ciphers in the reduced list to fail when establishing the connection. In this PR I returned the list of ciphers to the original list (before the hardening) if the user chose Trust any certificate = True (as done in [this PR](https://github.com/demisto/content/pull/19288) for Active Directory Query V2 Integration). 

4. Added a new TPB - LDAP Authentication - Test:
OpenLDAP_-_Test_Tue_Aug_30_2022 (1)

A  recommendation to use this integration for Active directory with STARTTLS was added to the detailed description of the 'Active Directory Authentication' (golang) integration - in [this MR](https://code.pan.run/xsoar/server/-/merge_requests/18450).

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [x] Yes
       - Further details: The new 'server type' parameter is required, but the default value is 'OpenLDAP' - so I think that users should be able to upgrade to this version without reconfiguring their existing instances. Not sure if it's considered a BC break. 
   - [ ] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by ShirleyDenkberg** [comment]
@JudahSchwartz Doc review completed.

**Review by JudahSchwartz — APPROVED** [review]
Amazing

File: Packs/OpenLDAP/Integrations/OpenLDAP/OpenLDAP.py
Comment (ShacharKidor) [reply Packs/OpenLDAP/Integrations/OpenLDAP/OpenLDAP.py:145]: Added - thanks :)

File: Packs/OpenLDAP/Integrations/OpenLDAP/OpenLDAP.py
Comment (ShacharKidor) [reply Packs/OpenLDAP/Integrations/OpenLDAP/OpenLDAP.py:31]: To deal with this issue in all the integrations that use HTTP communication, the CIPHERS_STRING was added to the CSP in the past (by @wissamg). 
Here we don't use HTTP, we use LDAP protocol and a raw socket - there are only two integrations that use the raw socket communication, this one, and the 'Active directory query V2' integration.
If you think it is necessary, we can add a base for them to the CSP - but I tend to leave it this way because it's only two integrations.

File: Packs/OpenLDAP/Integrations/OpenLDAP/OpenLDAP.py
Comment (JudahSchwartz) [inline Packs/OpenLDAP/Integrations/OpenLDAP/OpenLDAP.py:321]:
```suggestion
```
You did remove it =)

**Review by JudahSchwartz — APPROVED** [review]
Looks great. Lets wait until demo before merge

**Comment by ShacharKidor** [comment]
A link to a successful build that used the commit merge of a fix in the SDK - [3578632](https://code.pan.run/xsoar/content/-/pipelines/3578632).

**Comment by ShacharKidor** [comment]
@ShahafBenYakir - Please force merge the PR, the reasons are: 
1. Validate failed on a new required parameter - should be required. 
2. Lint failed on low coverage - I added unit test only for Active directory new methods due to this open issue - [CRTX-59131](https://jira-hq.paloaltonetworks.local/browse/CRTX-59131).

**Pull request merged**
